File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/admin/service/AdminStatsService.java
package io.github.hs96wings.streaming_server.admin.service;

import io.github.hs96wings.streaming_server.admin.dto.VisitStatDto;
import io.github.hs96wings.streaming_server.visitlog.repository.VisitLogRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.stream.Collectors;

@Service
@Transactional
public class AdminStatsService {
    private final VisitLogRepository visitLogRepository;

    public AdminStatsService(VisitLogRepository visitLogRepository) {
        this.visitLogRepository = visitLogRepository;
    }

    public List<VisitStatDto> getDailyStats() {
        List<Object[]> raw = visitLogRepository.countDailyUniqueVisitors();

        return raw.stream()
                .map(row -> new VisitStatDto(
                        ((java.sql.Date) row[0]).toLocalDate(),
                        (Long) row[1]
                ))
                .collect(Collectors.toList());
    }

}

File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/admin/service/AdminStatsService.kt
package io.github.hs96wings.streaming_server.admin.service

import io.github.hs96wings.streaming_server.admin.dto.VisitStatDto
import io.github.hs96wings.streaming_server.visitlog.repository.VisitLogRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.sql.Date
import java.util.stream.Collectors

@Service
@Transactional(readOnly = true)
class AdminStatsService(
    private val visitLogRepository: VisitLogRepository
) {
    fun getDailyStats(): List<VisitStatDto> {
        return visitLogRepository.countDailyUniqueVisitors().map{ (date, count) ->
            VisitStatDto(
                (date as java.sql.Date).toLocalDate(),
                count as Long
            )
        }
    }
}
